olive/evaluator/olive_evaluator.py

```diff
@@ -95,16 +107,19 @@ def _evaluate_custom(
         device: Device = Device.CPU,
         execution_providers=None,
     ) -> MetricResult:
-        # TODO: Change the evaluate function to accept the metric rather than
-        # breaking it into multiple arguments
-        # return eval_func(model, metric, dataloader, device, post_func)
-        raw_res = eval_func(
-            model,
-            get_local_path(metric.user_config.data_dir),
-            metric.user_config.batch_size,
-            device,
-            execution_providers,
-        )
+        raw_res = None
+        if metric.user_config.metric_func:
```

Comment:
I suppose the priority of evaluation function should be higher than that of metric function

Reply:
Updated.